### PR TITLE
Move corefxlab to arcade

### PR DESCRIPTION
**Pull request by pgovind**
For reviewers: I think you can ignore everything under `eng/common` and definitely everything related to `*/xlf/*`. That should only leave the `*.yml`, `*.props` and `*.csproj` files

cc: @eerhardt 

I'm still testing the OSX and Linux pipelines. I also need to remove the old build scripts that we're not using anymore. I'll remove the `WIP` tag once I'm done.

### CODE REVIEW

**Comment by ahsonkhan** [comment]
Are the xlf files related/necessary for this change or can we add them separately?

**Comment by pgovind** [comment]
> Are the xlf files related/necessary for this change or can we add them separately?

They are related, but I can add them separately. [This](https://github.com/dotnet/xliff-tasks) explains why I needed to check these changes in. I could make another patch with just the xlf changes. That would be a simple patch with nothing to review. I don't think anyone will look at the xlf files though? so all that'd do is just make 1 more PR. Up to you :)

**Comment by ahsonkhan** [comment]
> They are related, but I can add them separately. [This](https://github.com/dotnet/xliff-tasks) explains why I needed to check these changes in.

Why don't other repos like corefx/runtime need the xlf files then?

> I could make another patch with just the xlf changes. That would be a simple patch with nothing to review. I don't think anyone will look at the xlf files though? so all that'd do is just make 1 more PR. Up to you :)

When making cross-cutting changes like this (e.g. an infra update), it is generally better to keep the change isolated to only what is required. It makes it easier to revert if necessary (and makes it easier for someone else to follow/copy within a different context or repo). I don't fully understand why the xlf files are related to the arcade changes and if they are, doing it together is the right thing to do and makes sense. Otherwise, split it.

**Comment by eerhardt** [comment]
re: xlf files

Since `corefxlab` is an experimental repo, I don't see anyone actually taking the time (or spending the money) to localize any strings into any other languages. I would say let's opt out of putting these unnecessary files into the repo until they are used/necessary.

File: src/Directory.Build.props [hunk 1]
```diff
@@ -1,7 +1,3 @@
 <Project>
   <Import Project="..\Directory.Build.props" />
```
Comment (eerhardt) [inline src/Directory.Build.props]: This file can be removed. It doesn't provide any value.

File: eng/Signing.props [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<Project>
+  <!--
+    These are third party libraries that we use in Arcade. We need to sign them even if they
```
Comment (eerhardt) [inline eng/Signing.props]: Is this file necessary? Can it be removed?

File: eng/common-variables.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    # Publish-Build-Assets provides: MaestroAccessToken, BotAccount-dotnet-maestro-bot-PAT
+    # DotNet-HelixApi-Access provides: HelixApiAccessToken
+    - group: DotNet-Blob-Feed
+    - group: Publish-Build-Assets
+    - group: DotNet-HelixApi-Access
+    - group: SDL_Settings
+    - name: _InternalBuildArgs
+      value: /p:DotNetSignType=$(_SignType) /p:TeamName=$(_TeamName)
+        /p:DotNetPublishBlobFeedKey=$(dotnetfeed-storage-access-key-1)
+        /p:DotNetPublishBlobFeedUrl=https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json
```
Comment (eerhardt) [inline eng/common-variables.yml]: Do we want corefxlab's packages to be published to this blob feed? I don't think we mix corefxlab packages with our official packages, do we?

File: tools/dependencies.props [hunk 4]
```diff
@@ -18,6 +18,5 @@
     <MicroBuildCoreVersion>0.2.0</MicroBuildCoreVersion>
     <MicrosoftDotNetSignToolVersion>1.0.0-beta.19058.5</MicrosoftDotNetSignToolVersion>
     <MicrosoftSymbolUploaderBuildTaskVersion>1.0.0-beta-64414-01</MicrosoftSymbolUploaderBuildTaskVersion>
-    <MicrosoftSourceLinkVersion>1.0.0-beta2-19367-01</MicrosoftSourceLinkVersion>
```
Comment (eerhardt) [inline tools/dependencies.props]: Should `MicrosoftDotNetSignToolVersion` and `MicrosoftSymbolUploaderBuildTaskVersion` above be removed as well? Shouldn't we get those from arcade?

File: Directory.Build.props [hunk 5]
```diff
@@ -22,11 +22,12 @@
...
     <RepositoryUrl>https://github.com/dotnet/corefxlab</RepositoryUrl>
     <IncludeSymbols>true</IncludeSymbols>
+    <IsPackable>false</IsPackable>
+    <IsPackable Condition="$(MSBuildProjectName.StartsWith('System')) Or $(MSBuildProjectName.StartsWith('Microsoft'))">true</IsPackable>
   </PropertyGroup>
 
   <ItemGroup>
     <Content Include="$(RepoRoot)THIRD-PARTY-NOTICES.TXT" Pack="true" PackagePath="\" />
     <Content Include="$(RepoRoot)LICENSE" Pack="true" PackagePath="\" />
-    <Content Include="$(RepoRoot)tools/PackageIcon.png" Pack="true" PackagePath="$(PackageIcon)" Visible="false" />
```
Comment (eerhardt) [inline Directory.Build.props]: Can the file `tools/PackageIcon.png` be removed as well?

File: .azure-pipelines.yml [hunk 6]
```diff
@@ -11,35 +7,52 @@ trigger:
...
+  - template: \eng\common\templates\post-build\post-build.yml
     parameters:
-      name: Ubuntu1604
-      displayName: Innerloop Ubuntu16.04
-      targetOS: Ubuntu
-      pool:
-        name: Hosted Ubuntu 1604
+      validateDependsOn:
+        - build
+      enableSymbolValidation: false # https://github.com/dotnet/arcade/issues/2871
```
Comment (eerhardt) [inline .azure-pipelines.yml]: This issue has been closed for a few months. Do we still need this?

File: Directory.Build.targets [hunk 7]
```diff
@@ -1,4 +1,5 @@
 <Project>
+  <Import Project="Sdk.targets" Sdk="Microsoft.DotNet.Arcade.Sdk" />
```
Comment (safern) [inline Directory.Build.targets]: After importing arcade we might be able to remove the workaround to translate the url for source link since that is already done in Arcade.

File: build.sh [hunk 8]
```diff
@@ -1,123 +1,16 @@
...
-  do
-    echo -e "${RED}    $i${NC}"
-  done
-else
-  echo -e "${GREEN}** Build succeeded. **${NC}"
-fi
-
-exit $errorsEncountered
+scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
+"$scriptroot/eng/common/build.sh" --build --restore $@
```
Comment (safern) [inline build.sh]: Nit: extra line at EOF

File: build.cmd [hunk 9]
```diff
@@ -1,3 +1,3 @@
 @echo off
-setlocal
-powershell -NoProfile %~dp0scripts\build.ps1 %*
\ No newline at end of file
+powershell -ExecutionPolicy ByPass -NoProfile -command "& """%~dp0eng\common\Build.ps1""" -restore -build %*"
+exit /b %ErrorLevel%
```
Comment (safern) [inline build.cmd]: Nit: extra line at EOF.

File: eng/Signing.props [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<Project>
```
Comment (safern) [inline eng/Signing.props]: Do we really need this empty file? It is only imported when it exists: https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/Sign.proj#L48

File: eng/common-variables.yml [hunk 11]
```diff
@@ -0,0 +1,39 @@
+variables:
```
Comment (safern) [inline eng/common-variables.yml]: Is this needed? It seems to not be imported by corefxlab's pipeline.

File: .gitignore [hunk 12]
```diff
@@ -269,3 +269,7 @@ binaries/
 
 # Do not include launch settings
 launchSettings.json
+
+.dotnet/
+artifacts/
+/.packages
```
Comment (safern) [inline .gitignore]: You should also add folders from eng/common that you don't need/care about, like `cross` , `internal` and `native` those are directories that you will not use and I don't imagine corefxlab using it, so I think is better to exclude them and if needed we can add them later on.

File: eng/common-variables.yml (hunk 3 above)
Comment (safern) [reply eng/common-variables.yml]: I think this file is not even imported and we can just delete it since it is confusing. This is the real blob feed we're publishing to: https://github.com/dotnet/corefxlab/pull/2795/files#diff-4eda78ea0bc35794fd331e0d76b7872fR40

we created an experimental one in order to avoid mixing corefxlab with official packages.

File: eng/Signing.props (hunk 2 above)
Comment (pgovind) [reply eng/Signing.props]: Removed

File: scripts/build.ps1 [hunk 13]
```diff
@@ -1,100 +0,0 @@
...
-Write-Host "Version=$Version."
-Write-Host "BuildVersion=$BuildVersion."
-Write-Host "SkipTests=$SkipTests."
-
-$dotnetExePath="$PSScriptRoot\..\dotnetcli\dotnet.exe"
-if ($Version -eq "<default>") {
-    $Version = (Get-Content "$PSScriptRoot\..\DotnetCLIVersion.txt" -Raw).Trim()
-}
-
-if (!(Test-Path "$PSScriptRoot\..\dotnetcli")) {
```
Comment (ahsonkhan) [inline scripts/build.ps1]: With us reworking the corefxlab build scripts/infra, can you please explain how contributors build/test corefxlab now, what dependencies (if any) are needed, and more importantly how to use the packages?

We should update the README accordingly, specifically these sections:
https://github.com/dotnet/corefxlab#how-to-use - This still mentions the use of the MyGet feed
https://github.com/dotnet/corefxlab#building-and-testing - This currently explains where the CLI is downloaded, how to set the path, and how the tests for the whole corefxlab.sln work.

File: build.sh [hunk 14]
```diff
@@ -1,123 +1,16 @@
...
-Restore="true"
-echo "Configuration=$Configuration."
-echo "Restore=$Restore."
-echo "Version=$Version."
-echo "BuildVersion=$BuildVersion."
-
-dotnetExePath="dotnetcli/dotnet"
-
-if [ "$Version" = "<default>" ]; then
-  Version=$(head -n 1 "DotnetCLIVersion.txt")
```
Comment (ahsonkhan) [inline build.sh]: How do we now control/update the CLI version we need to install/use in corefxlab?

Also, is the plan to remove DotnetCLIVersion.txt?
https://github.com/dotnet/corefxlab/blob/154fafa095dd6159867c4aeb096d31c60ae96ad0/DotnetCLIVersion.txt#L1

File: archived_projects/tests/System.CommandLine.Tests/System.CommandLine.Tests.csproj [hunk 15]
```diff
@@ -14,11 +14,6 @@
     <PackageLicenseUrl></PackageLicenseUrl>
     <PackageRequireLicenseAcceptance>False</PackageRequireLicenseAcceptance>
   </PropertyGroup>
-  <ItemGroup>
```
Comment (ahsonkhan) [inline archived_projects/tests/System.CommandLine.Tests/System.CommandLine.Tests.csproj]: Is it necessary to modify the `archived_projects`?

File: build.sh [hunk 16]
```diff
@@ -1,123 +1,16 @@
...
+done
 
-if [ $# -eq 0 ]
-  then
-    Configuration="Debug"
-  else
-    Configuration=$1
-fi
-
-BuildVersion="preview2-$(date '+%y%m%d')-1"
```
Comment (ahsonkhan) [inline build.sh]: What are the package versioning rules now? Can we control it?

File: eng/common/performance/performance-setup.ps1 [hunk 17]
```diff
@@ -0,0 +1,106 @@
...
+. "$PSScriptRoot\..\pipeline-logging-functions.ps1"
+
+# Directories
+Write-PipelineSetVariable -Name 'PayloadDirectory' -Value "$PayloadDirectory" -IsMultiJobVariable $false
+Write-PipelineSetVariable -Name 'PerformanceDirectory' -Value "$PerformanceDirectory" -IsMultiJobVariable $false
+Write-PipelineSetVariable -Name 'WorkItemDirectory' -Value "$WorkItemDirectory" -IsMultiJobVariable $false
+
+# Script Arguments
+Write-PipelineSetVariable -Name 'Python' -Value "py -3" -IsMultiJobVariable $false
+Write-PipelineSetVariable -Name 'ExtraBenchmarkDotNetArguments' -Value "$ExtraBenchmarkDotNetArguments" -IsMultiJobVariable $false
```
Comment (ahsonkhan) [inline eng/common/performance/performance-setup.ps1]: Do we need all the performance related scripts?

File: build.cmd [hunk 18]
```diff
@@ -1,3 +1,3 @@
 @echo off
```
Comment (ahsonkhan) [inline build.cmd]: Does `clean.cmd` still work as is or do we need to update it?
https://github.com/dotnet/corefxlab/blob/154fafa095dd6159867c4aeb096d31c60ae96ad0/clean.cmd#L6-L11

File: tests/Benchmarks/Benchmarks.csproj [hunk 19]
```diff
@@ -31,6 +31,7 @@
     <PackageReference Include="BenchmarkDotNet.Diagnostics.Windows" Version="$(BenchmarkDotNetVersion)" />
     <PackageReference Include="Microsoft.Diagnostics.Tracing.TraceEvent" Version="2.0.30" />
     <PackageReference Include="System.Text.Encodings.Web" Version="$(CoreFxStableVersion)" />
+    <PackageReference Include="xunit" Version="$(XUnitVersion)" IsImplicitlyDefined="true" PrivateAssets="all" Publish="true"/>
```
Comment (ahsonkhan) [inline tests/Benchmarks/Benchmarks.csproj]: Can you please explain what these addition properties do and why they are needed (`IsImplicitlyDefined="true" PrivateAssets="all" Publish="true"`)?

File: .azure-pipelines.yml (hunk 6 above)
Comment (joperezr) [reply .azure-pipelines.yml]: Even though you are right(the issue has been closed for a while), Most repos still have this as I believe we haven't yet figured out a way to do symbol validation correctly. Here is a pointer to the Arcade repo which has today in master the exact same clause:

https://github.com/dotnet/arcade/blob/master/azure-pipelines.yml#L181-L183

File: .gitignore [hunk 20]
```diff
@@ -269,3 +269,14 @@ binaries/
...
+eng/common/cross/
+eng/common/internal/
+eng/common/native/
+eng/common/performance/
+eng/common/post-build/
+eng/common/sdl/
+eng/common/templates/
```
Comment (joperezr) [inline .gitignore]: Are you sure you don't need templates? I would imagine that those plus the post-build are important, so you probably want to remove those from here.

File: build.sh (hunk 14 above)
Comment (joperezr) [reply build.sh]: Same as in dotnet/runtime and the rest of the arcade repos, this will be controlled via the root's global.json file.

Comment (joperezr) [reply build.sh]: > Also, is the plan to remove DotnetCLIVersion.txt?

Yes that file should be removed.

File: src/Directory.Build.props (hunk 1 above)
Comment (joperezr) [reply src/Directory.Build.props]: @pgovind beware that if you do endup removing this file, then you will need to update all of the projects that were importing this in order to import the root props file.

File: Directory.Build.targets [hunk 21]
```diff
@@ -17,19 +18,4 @@
   <PropertyGroup>
     <Version Condition="'$(IsStableRelease)' == 'true' and '$(IsShipping)' == 'true'">$(VersionPrefix)</Version>
```
Comment (joperezr) [inline Directory.Build.targets]: I would expect both sourcelink and version info to be all calculated by arcade instead of this repo, so idieally you should be able to delete all this.

File: eng/common/native/common-library.sh [hunk 22]
```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env bash
+
+function GetNativeInstallDirectory {
```
Comment (joperezr) [inline eng/common/native/common-library.sh]: As @safern mentioned, you probably won't need this folder or the cross one, so don't check them in for now and add them to the git ignore. Anything checked in should not be present on that git ignore, and anything that you do need and is checked should NOT be in the git ignore.

File: global.json [hunk 23]
```diff
@@ -0,0 +1,14 @@
+{
+  "tools": {
+    "dotnet": "3.0.100",
+    "runtimes": {
+      "dotnet": [
+        "2.1.11"
```
Comment (joperezr) [inline global.json]: Do we actually need this or did you copy it from a different repo? This is only sometimes needed depending on the frameworks targeted by the projects, so it may not be needed in your case.

**Comment by joperezr** [comment]
>     <add key="dotnet.myget.org dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" protocolVersion="3" />

You should get rid of all of these myget feeds and only depend on the blob feeds instead.

---
Refers to: NuGet.Config:20 in 0849f88. [](commit_id = 0849f88a5cb258c1512691cc451a20b5bc26d1dd, deletion_comment = False)

File: NuGet.Config [hunk 24]
```diff
@@ -28,5 +28,6 @@
     <add key="nugetbuild" value="https://www.myget.org/F/nugetbuild/api/v3/index.json" />
     <add key="roslyn" value="https://dotnet.myget.org/F/roslyn/" />
     <add key="benchmarkdotnet-ci" value="https://ci.appveyor.com/nuget/benchmarkdotnet" />
+    <add key="arcade" value="https://dotnetfeed.blob.core.windows.net/dotnet-tools-internal/index.json" />
```
Comment (joperezr) [inline NuGet.Config]: @mmitche can correct me if I'm wrong, but I believe that here you may want to use this feed instead: 
```xml
<add key="dotnet-tools" value="[https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json](https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json)" />
```

File: .azure-pipelines.yml (hunk 6 above)
Comment (eerhardt) [reply .azure-pipelines.yml]: @JohnTortugo - should this URL be linked somewhere else? It doesn't seem right for us to add a new setting with a link to a closed issue.

File: NuGet.Config (hunk 24 above)
Comment (mmitche) [reply NuGet.Config]: Actually:
```
<add key="dotnet-eng" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json" />
```

File: build.cmd (hunk 18 above)
Comment (safern) [reply build.cmd]: We should remove `clean.cmd` and people should just use build -clean... which will remove the artifacts directory.

File: tests/Benchmarks/Benchmarks.csproj (hunk 19 above)
Comment (safern) [reply tests/Benchmarks/Benchmarks.csproj]: `IsImpicitlyDefined` is so that you can't change the version in the editor. `Publish=true` tells the SDK to include it in the publish folder because we're marking all the assets as `Private`. `PrivateAssets=all` is to tell the SDK to not flow this dependency as a transitive dependency to projects that take a dependency on this project or package.

File: scripts/build.ps1 (hunk 13 above)
Comment (joperezr) [reply scripts/build.ps1]: +1. You should be able to mostly copy paste from the rest of our repos’ Readmes since we are now going to have very similar instructions to the basic repos

File: build.sh (hunk 16 above)
Comment (joperezr) [reply build.sh]: We could if we wanted to, but the way this is will now use the same arcade versioning that the rest of the repos use. That is (version)-(prereleaselabel)-(prereleaseRevision)-(arcadegeneratedDateBasedVersion).(BuildRevision)

Comment (joperezr) [reply build.sh]: We could if we wanted to, but the way this is will now use the same arcade versioning that the rest of the repos use. That is (version)-(prereleaselabel)-(prereleaseRevision)-(arcadegeneratedDateBasedVersion).(BuildRevision)

Comment (ahsonkhan) [reply build.sh]: Will the versions always remain in pre-release/preview? I am asking to make sure we avoid marking them as stable.

File: build.cmd (hunk 18 above)
Comment (ahsonkhan) [reply build.cmd]: Sounds good. Let's remove it.

I will copy this script to my dev box instead, since I use it to close all dotnet.exe instances running (I don't think `build -clean` does that).

File: tests/Benchmarks/Benchmarks.csproj (hunk 19 above)
Comment (ahsonkhan) [reply tests/Benchmarks/Benchmarks.csproj]: Thanks for the explanation, @safern 

> `IsImpicitlyDefined` is so that you can't change the version in the editor.

And why do we want to disallow that, for the xunit package reference, specifically?

File: build.sh (hunk 16 above)
Comment (safern) [reply build.sh]: Yes. You need to specify a property to mark as stable and we can also do that on a per project basis.

File: tests/Benchmarks/Benchmarks.csproj (hunk 19 above)
Comment (safern) [reply tests/Benchmarks/Benchmarks.csproj]: We usually do that for our tooling specially since it is a shared version across different projects. This is using `XunitVersion` msbuild property. I bet that if we allowed that, VS would change `XUnitVersion` for the actual textual version, so then this project wouldn't use the shared property.

**Comment by pgovind** [comment]
I'll address all the comments here soon. Need to get some other stuff merged first.

File: build.cmd (hunk 18 above)
Comment (safern) [reply build.cmd]: We can of course propose that into arcade... it would be handy if arcade searched for dotnet.exe and killed it.

**Comment by safern** [comment]
@pgovind this seems to be publishing test projects as nuget packages as well. Can you please fix that?

https://dev.azure.com/dnceng/public/_packaging?_a=feed&feed=dotnet-experimental

File: tools/dependencies.props (hunk 4 above)
Comment (pgovind) [reply tools/dependencies.props]: Testing with those 2 commented out.

**Comment by pgovind** [comment]
CI keeps getting a permission denied error: 
./build.sh: line 16: /home/vsts/work/1/s/eng/common/build.sh: Permission denied

I remember reading somewhere that this is because of copying arcade into the folder on windows instead of cloning it, but I can't seem to find that documentation anymore. Just pinging @joperezr or @safern to see if they've seen this before while I search for a *nix machine to check build.sh's flags.

File: eng/common/performance/performance-setup.ps1 (hunk 17 above)
Comment (pgovind) [reply eng/common/performance/performance-setup.ps1]: Nope. Deleted.

Comment (pgovind) [reply eng/common/performance/performance-setup.ps1]: Nope, deleted.

**Comment by eerhardt** [comment]
> CI keeps getting a permission denied error:
./build.sh: line 16: /home/vsts/work/1/s/eng/common/build.sh: Permission denied

https://stackoverflow.com/questions/21691202/how-to-create-file-execute-mode-permissions-in-git-on-windows

File: tools/dependencies.props (hunk 4 above)
Comment (pgovind) [reply tools/dependencies.props]: Removed

File: Directory.Build.targets (hunk 21 above)
Comment (pgovind) [reply Directory.Build.targets]: I think this is what overrides the version number when we set the `IsStableRelease` variable though.

File: global.json (hunk 23 above)
Comment (pgovind) [reply global.json]: Testing with the runtime not specified.

File: eng/Versions.props [hunk 25]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <VersionPrefix>1.0.0</VersionPrefix>
```
Comment (bartonjs) [inline eng/Versions.props]: Which one wins, this one, or the root Directory.Build.props? (My guess is that the root one wins, because I think it includes this then over-writes it).  Ideally this value is only defined once.

File: eng/Versions.props [hunk 26]
```diff
@@ -0,0 +1,8 @@
...
+    <PreReleaseVersionLabel>prerelease</PreReleaseVersionLabel>
```
Comment (bartonjs) [inline eng/Versions.props]: Oh, joy, we're changing the label again.  Didn't cause any problems last time (with a bystander effect of everyone assuming someone else would fix their packages :smile:).

Comment (joperezr) [reply eng/Versions.props]: packages will go to a new feed with these changes, and their entire versioning label logic will change as well, so consumers will have to react to changes no matter what. @safern just suggested (offline) to use alpha instead since they are new and we might in the future want to ship beta packages that sort greater than alpha.

File: eng/Versions.props (hunk 25 above)
Comment (joperezr) [reply eng/Versions.props]: it really depends on when the imports are happening. I agree with the suggestion that this should really only be defined once, and I would keep this one as this file should contain all the versioning properties, so I would remove the one from Directory.Build.props

File: eng/Versions.props (hunk 26 above)
Comment (bartonjs) [reply eng/Versions.props]: Assuming the package names are staying the same (which seems legit), we should ensure that the versions are monotonic.  Which may just mean bumping to 0.1.3.

I don't think that we should jump from 0.1 to 1.0 just because we moved to arcade.

Comment (joperezr) [reply eng/Versions.props]: because they are going to a different feed we might not even need the bump. We'll just tell people to reference the latest version of the packages on the new feed and that should be enough.

Comment (bartonjs) [reply eng/Versions.props]: > We'll just tell people to reference the latest version of the packages on the new feed and that should be enough.

I don't know if the nuget package cache considers the same name in two feeds to be different packages, or the same.  I advise assuming "same" and make sure that we don't get cache-related confusion by making sure that all new-feed packages have higher package versions than all myget-feed packages with the same name.

Comment (joperezr) [reply eng/Versions.props]: It is true cache will treat them as same, but the prerelease label format  is changing as well, we used to have some special repo specific format for corefxlab packages but we are changing that to the Arcade format with this which is different and should have the packages never clash with existing ones on your cache. We can keep this in mind but the cache in theory shouldn't get in the way of figuring out which package is the latest assuming that you remove the old myget feed from your project and you now use the blob feed, even in the case where there are myget package caches on your machine.

File: eng/Versions.props (hunk 25 above)
Comment (pgovind) [reply eng/Versions.props]: Done. Defined only in `eng/Versions.props` now. Also bumped the version by incrementing the patch by 1 and renamed the packages to have `alpha` now instead of `preview`.

**Comment by pgovind** [comment]
I'm waiting to see the artifacts produced here(https://dev.azure.com/dnceng/public/_packaging?_a=feed&feed=dotnet-experimental) and verify their version numbers. Once that's done, this should be ready for review.

File: .gitignore [hunk 27]
```diff
@@ -269,3 +269,11 @@ binaries/
 
 # Do not include launch settings
 launchSettings.json
+
+.dotnet/
+artifacts/
+/.packages
+eng/common/cross/
+eng/common/native/
+eng/common/performance/
```
Comment (joperezr) [inline .gitignore]: Obviously people can add this in the future if they need to, but perhaps we should keep performance? Usually stuff that goes in corefxlab is for testing and see if it is something we want to move to runtime repo, so perhaps being able to run performance tests is important.

File: corefxlab-base.yml [hunk 28]
```diff
@@ -0,0 +1,78 @@
...
+
+          - ${{ if eq(parameters.isOfficialBuild, 'false') }}:
+            - name: _buildArguments
+              value: -Configuration $(_BuildConfig)
+
+        pool: ${{ parameters.pool }}
+
+        strategy:
+          matrix:
+            ${{ if eq(parameters.isOfficialBuild, 'false') }}:
```
Comment (joperezr) [inline corefxlab-base.yml]: NIT: perhaps put x64_Release one first since that one is always built and have the conditioned one after :smile:

File: eng/Versions.props [hunk 29]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <VersionPrefix>0.1.2</VersionPrefix>
+    <PreReleaseVersionLabel>alpha</PreReleaseVersionLabel>
+    <UsingToolXliff>false</UsingToolXliff>
```
Comment (joperezr) [inline eng/Versions.props]: NIT: I'm not sure if UsingToolXliff should go in this file, this feels more like something that should go in Directory.Build.props.

File: global.json [hunk 30]
```diff
@@ -0,0 +1,9 @@
+{
+  "tools": {
+    "dotnet": "3.0.100"
```
Comment (joperezr) [inline global.json]: might as well just use the latest one while you are at it "3.1.101"

File: src/Directory.Build.props (hunk 1 above)
Comment (joperezr) [inline src/Directory.Build.props]: Just a reminder that you can remove this file if you wanted too. Again, if you do just make sure that you make sure people upstack are importing the right Directory.Build.props from the root.

File: tests/Benchmarks/Benchmarks.csproj (hunk 19 above)
Comment (joperezr) [inline tests/Benchmarks/Benchmarks.csproj]: Make sure we still need this, as I believe that because you set IsTestProject on all projects under tests, I would imagine this reference should be fetched automatically.

File: tools/dependencies.props [hunk 31]
```diff
@@ -16,8 +16,5 @@
     <SnVersion>1.0.0</SnVersion>
     <VSWhereVersion>2.5.2</VSWhereVersion>
     <MicroBuildCoreVersion>0.2.0</MicroBuildCoreVersion>
-    <MicrosoftDotNetSignToolVersion>1.0.0-beta.19058.5</MicrosoftDotNetSignToolVersion>
```
Comment (joperezr) [inline tools/dependencies.props]: Note that in this file you have a bunch of versions for corefx packages that should probably instead be comminng from an arcade subscription in order to guarantee that corefxlab always has the latest version of those dependencies. You don't have to fix that now, but might be something to think about in the future.

File: .gitignore (hunk 27 above)
Comment (safern) [reply .gitignore]: I would leave it out at the moment. This are mostly yml files and scripts to run performance tests that live on the perf repo. I think we can just remove it whenever we enable those perf builds.

File: src/Directory.Build.props (hunk 1 above)
Comment (pgovind) [reply src/Directory.Build.props]: Right, that's why I'm leaving it here. Otherwise I'd have to update each csproj to refer to the root Directory.Build.props. This way any new proj added to the repo should just inherit the props

File: tests/Benchmarks/Benchmarks.csproj (hunk 19 above)
Comment (pgovind) [reply tests/Benchmarks/Benchmarks.csproj]: Testing without this ref. Don't remember why this was needed anymore.

File: tools/dependencies.props (hunk 31 above)
Comment (pgovind) [reply tools/dependencies.props]: Good point. Testing with all of these removed.

Comment (safern) [reply tools/dependencies.props]: Actually those need to stay like that. @ahsonkhan left them like that in purpose, we don't want a subscription for those just yet, right @ahsonkhan?

Also, this file should be eng/Versions.props no? That's imported by default by arcade.

Comment (pgovind) [reply tools/dependencies.props]: Yup, it needed all those version numbers. I did move it to eng/Versions.props though.

File: tools/dependencies.props [hunk 32]
```diff
@@ -1,20 +1,4 @@
 <Project>
```
Comment (safern) [inline tools/dependencies.props]: Make sure to remove this file before merging.

Comment (pgovind) [reply tools/dependencies.props]: Yup yup. Removed the file now. I was just testing locally if the build will fail when there are references to this file and I remove it. I removed the references to the file now too and everything should succeed.

**Comment by pgovind** [comment]
Alright, this is ready to merge. I verified that the internal build is fine and publishes the right packages and versions. I will also be adding a new project to corefxlab soon, so if we've missed something, we will hit it then. Huge thanks to @safern and @joperezr for helping me put this up!

**Comment by safern** [comment]
Thanks @pgovind this is great!
